packages/muelu/src/Misc/MueLu_CoordinatesTransferFactory_def.hpp

```diff
@@ -202,59 +201,109 @@ namespace MueLu {
 
       RCP<const Map>   uniqueMap      = fineCoords->getMap();
       RCP<const Map>   coarseCoordMap = MapFactory        ::Build(coarseMap->lib(), Teuchos::OrdinalTraits<Xpetra::global_size_t>::invalid(), elementList, indexBase, coarseMap->getComm());
-      coarseCoords   = Xpetra::MultiVectorFactory<typename Teuchos::ScalarTraits<Scalar>::coordinateType,LO,GO,NO>::Build(coarseCoordMap, fineCoords->getNumVectors());
+      coarseCoords   = Xpetra::MultiVectorFactory<typename Teuchos::ScalarTraits<Scalar>::magnitudeType,LO,GO,NO>::Build(coarseCoordMap, fineCoords->getNumVectors());
+
+
+      RCP<Aggregates>        aggregates;
+      RCP<Aggregates_kokkos> aggregatesKokkos;
+      bool aggregatesCrossProcessors;
+      if (IsType<RCP<Aggregates> >(fineLevel, "Aggregates")) {
+        aggregates = Get< RCP<Aggregates> > (fineLevel, "Aggregates");
+        aggregatesCrossProcessors = aggregates->AggregatesCrossProcessors();
+      } else {
+        aggregatesKokkos = Get<RCP<Aggregates_kokkos> >(fineLevel, "Aggregates");
+        aggregatesCrossProcessors = aggregatesKokkos->AggregatesCrossProcessors();
+      }
 
       // Create overlapped fine coordinates to reduce global communication
       RCP<xdMV> ghostedCoords = fineCoords;
-      if (aggregates->AggregatesCrossProcessors()) {
+      if (aggregatesCrossProcessors) {
```

Comment:
We should really just get rid of coupled coarsening.  It's slow, afaik we don't ever use it, and it would improve code maintainability.

Reply:
Maybe that's a good first step before we start merging more aggregation related Kokkos things?